modules/light-clients/08-wasm/types/mock_engine_test.go

```diff
@@ -94,10 +116,14 @@ func (m *MockWasmEngine) Query(codeID wasmvm.Checksum, env wasmvmtypes.Env, quer
 
 // Sudo implements the WasmEngine interface.
 func (m *MockWasmEngine) Sudo(codeID wasmvm.Checksum, env wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
-	if m.SudoFn == nil {
-		panic("mock engine is not properly initialized")
+	msgTypeName := getQueryMsgPayloadTypeName(sudoMsg)
+
+	sudoFn, ok := m.sudoCallbacks[msgTypeName]
+	if !ok {
+		panic(fmt.Errorf("no callback specified for %s", msgTypeName))
```

Comment:
It'd be nice if we can change the error message here as you do register callbacks for all expected sudo messages in `NewMockWasmEngine`. So maybe this should be something like unexpected argument type?

Alternatively, you can change the default panic message in line 50 to something like `"default callback panic for msgTypeName"`

Reply:
talked in slack, can do these refactors in follow ups